src/relay/frontend/build_module.h

```diff
@@ -0,0 +1,355 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file relay/backend/compile_engine.h
+ * \brief Internal compilation engine handle function cache.
+ *  and interface to low level code generation.
+ */
+#ifndef TVM_RELAY_FRONTEND_BUILD_MODULE_H_
+#define TVM_RELAY_FRONTEND_BUILD_MODULE_H_
+#include <vector>
+#include <string>
+
+#include "utils.h"
+#include "graph_runtime_codegen.h"
+#include "tvm_build_module.h"
+
+namespace tvm {
+namespace relay {
+namespace frontend {
+
+/*! \brief Optimization pass level */
+struct OPT_PASS_LEVEL_DECL {
+  const std::unordered_map<std::string, char> _data {
```

Comment:
make this static? Also why not std::unordered_map<std::string, size_t>? Seems weird having this as a `char` instead of a logical integer right?

Reply:
Yeah I will make it static. As we only need 0-3 for level, char is enough to represent all levels. But this is minor we can use size_t.